test_runner/regress/test_compute_reconfigure.py

```diff
@@ -85,3 +87,58 @@ def check_logs():
     samples = metrics.query_all("compute_ctl_up", {"build_tag": build_tag})
     assert len(samples) == 1
     assert samples[0].value == 1
+
+
+def test_compute_safekeeper_connstrings_duplicate(neon_simple_env: NeonEnv):
+    """
+    Test that we can change postgresql.conf settings even if
+    skip_pg_catalog_updates=True is set.
+    """
+    env = neon_simple_env
+
+    endpoint = env.endpoints.create_start("main")
+
+    # grab the current value of neon.safekeepers
+    sk_list = []
+    with endpoint.cursor() as cursor:
+        cursor.execute("SHOW neon.safekeepers;")
+        row = cursor.fetchone()
+        assert row is not None
+
+        log.info(f'    initial neon.safekeepers: "{row}"')
+
+        # build a safekeepers list with a duplicate
+        sk_list.append(row[0])
+        sk_list.append(row[0])
+
+    safekeepers = ",".join(sk_list)
+    log.info(f'reconfigure neon.safekeepers: "{safekeepers}"')
+
+    # introduce duplicate entry in neon.safekeepers, on purpose
+    endpoint.respec_deep(
+        **{
+            "spec": {
+                "skip_pg_catalog_updates": True,
+                "cluster": {
+                    "settings": [
+                        {
+                            "name": "neon.safekeepers",
+                            "vartype": "string",
+                            "value": safekeepers,
+                        }
+                    ]
+                },
+            },
+        }
+    )
+
+    try:
+        endpoint.reconfigure()
```

Comment:
Re testing, I was thinking about teaching reconfigure() in test fixtures and neon_local cli to return http status code and error (if it happens). That way you can handle it gracefully in test and check returned code and error

Reply:
To be fair, it sounds to me like that would be a separate PR. I kind of expected errors transmission up the call-chain to already be in place in the context of this PR.